src/main/java/domain/LottoStatisticData.java

```diff
@@ -0,0 +1,81 @@
+package domain;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+public class LottoStatisticData {
+    private final LottoNumbers winnerNumbers;
+    private final LottoNumber bonusNumber;
+    private final HashMap<Match, Integer> matchStatistic;
+
+    public LottoStatisticData(LottoNumbers winnerNumbers, LottoNumber bonusNumber, List<Lotto> purchasedLottos) {
+        this.winnerNumbers = winnerNumbers;
+        this.bonusNumber = bonusNumber;
+        this.matchStatistic = compileLottosStatistic(purchasedLottos);
+    }
+
+    private HashMap<Match, Integer> compileLottosStatistic(List<Lotto> lottos) {
+        HashMap<Match, Integer> matchCounts = initializeMatchMap();
+
+        for (Lotto lotto : lottos) {
+            List<LottoNumber> lottoNumbers = lotto.getLottoNumbers().getNumbers();
+
+            int cnt = findMatchNumberCount(lottoNumbers);
+
+            boolean bonusFlag = isBonusMatched(cnt, lottoNumbers);
+
+            upDateMatchStatistic(cnt, bonusFlag, matchCounts);
+        }
+
+        return matchCounts;
+    }
+
+    private int findMatchNumberCount(List<LottoNumber> lottoNumbers) {
+        int cnt = 0;
+
+        for (LottoNumber lottoNumber : winnerNumbers.getNumbers()) {
+            if (lottoNumbers.contains(lottoNumber)) {
+                cnt++;
+            }
+        }
+
+        return cnt;
+    }
+
+    private boolean isBonusMatched(int cnt, List<LottoNumber> lottoNumbers) {
+        return cnt == 5 && lottoNumbers.contains(bonusNumber);
+    }
+
+    private void upDateMatchStatistic(int cnt, boolean bonusFlag, HashMap<Match, Integer> matchCounts) {
+        Match match = Match.from(cnt, bonusFlag);
+
+        if (match != null) {
+            matchCounts.put(match, matchCounts.get(match) + 1);
+        }
+    }
+
+
+
+    private HashMap<Match, Integer> initializeMatchMap() {
+        HashMap<Match, Integer> matchCounts = new HashMap<Match, Integer>();
+
+        Arrays.stream(Match.values())
+                .forEach(v -> matchCounts.put(v, 0));
+
+        return matchCounts;
+    }
```

Comment:
Stream을 활용하면 더 줄여볼 수 있어요.
```suggestion
    private Map<Match, Integer> initializeMatchMap() {
        return Arrays.stream(Match.values())
                .collect(Collectors.toMap(v -> v, v -> 0));
    }
```